### ISSUE TITLE
Logging Problem

### ISSUE THREAD

**Issue by glacroix** [sipwise/rtpengine#1341]
Hello,

We use i RTPEngine 9.5 with Kamailio. For some calls, the logs get  flooding with the following message : 

> rtpengine[1014]: ERR: [02385878-bd21-1910-9db6-005056950de1@call port 36874]: [transcoding] Switched from input to output codec context, but no codec handler present

The sound works perfectly nevertheless.

Do you have any clue why ? Here is the call to RTP Engine from Kamailio :

> rtpengine_manage("RTP/AVP codec-mask-all codec-transcode-PCMU codec-transcode-PCMA codec-transcode-G729 codec-transcode-GSM direction=internal direction=external replace-origin replace-session-connection SIP-source-address ICE=remove");

The logs get very big pretty fast, so you have any solution...

Thanks for your help,
Guillaume

**Comment by rfuchs** [comment]
Post a full debug log for such a call

**Comment by glacroix** [comment]
You mean a log-level to 6 ? Actually, the logs are filled so fast that it is hard to see anyhtiong (Gb in few time). I have found this online: [https://www.gitmemory.com/issue/sipwise/rtpengine/1259/834378386](url)

I dont know if that could help.

**Comment by rfuchs** [comment]
Yes I'm aware of this (it's my own answer). Does your version have the log limiter for this log line or do you need to update? (`9.5` isn't a single specific version)

**Comment by glacroix** [comment]
Right, thank you ! It handles max 40 concurrent calls.
Here is a full capture of a line :

> vpn-sip-bridge-th2-1 rtpengine[1014]: ERR: [e37cf9c4-bd21-1910-8fa3-0050569508fd@call port 49588]: [transcoding] Switched from input to output codec context, but no codec handler present

How can I check the storage limit ?

**Comment by rfuchs** [comment]
40 calls isn't that much really. High CPS then? The log limiter holds 10k entries so that shouldn't be a problem. Does each log line actually repeat?

**Comment by glacroix** [comment]
No, CPS must be between 2 and 3 max. 
Yes, it is repeated lots of time so MB become GB pretty fast. I had to set the log level to 1 to stop it, but not a good solution as we need some Error logs.

<img width="1085" alt="Capture d’écran 2021-09-03 à 19 40 00" src="https://user-images.githubusercontent.com/5825341/132046035-5a0a9054-29b8-47eb-b2b3-b1d11daa2294.png">

**Comment by rfuchs** [comment]
Post your exact version then because this shouldn't happen on 9.5 with such a low volume of calls.

**Comment by glacroix** [comment]
I have done the install from this file :
ngcp-rtpengine_9.5.0.0+0~mr9.5.0.0_all.deb

**Comment by rfuchs** [comment]
That's not a properly tagged version so might not have the log line rate limited. Maybe try an update?

**Comment by glacroix** [comment]
I will have to find a moment with no traffic (it has been switched to production). In the meantime I looked in the code and found agai: 
Version:	9.5.0.0+0~mr9.5.0.0

**Comment by glacroix** [comment]
We have not yet updated our version, but we have removed the transcoding options ('codec-mask-all codec-transcode-PCMU codec-transcode-PCMA codec-transcode-G729 codec-transcode-GSM') and the problem disappears. Maybe we have just missed adding a 'transcode-telephone-event' tag as you mentionned the problem is bound to DTMF...
